### PR TITLE
Run conformance test for combinations of consul + consul-k8s

### PR DESCRIPTION
**Pull request by nathancoleman**
### Changes proposed in this PR:
Add a test matrix for conformance tests verifying that API Gateway works w/ consul 1.11.x in addition to consul 1.12.x

### How I've tested this PR:
🤖 tests pass

### How I expect reviewers to test this PR:
🤖 tests pass

### Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: .github/workflows/conformance.yml [hunk 1]
```diff
@@ -95,7 +108,7 @@ jobs:
       - name: Install Consul
         working-directory: "consul-api-gateway/internal/testing/conformance"
         run: |
-          helm install --values ./consul-config.yaml consul $GITHUB_WORKSPACE/consul-k8s/charts/consul --create-namespace --namespace=consul
+          helm install --values ./consul-config.yaml consul hashicorp/consul --version ${{ matrix.config.consul-k8s-version }} --set global.image=${{ matrix.config.consul-image }} --set global.imageEnvoy=${{ matrix.config.envoy-image }} --create-namespace --namespace=consul
```
Comment (nathancoleman) [inline .github/workflows/conformance.yml:105]: cloning `main` has been breaking frequently, and cloning the tag when we can just install from the repo doesn't make sense

**Review by nathancoleman — COMMENTED** [review]
Personal Review

File: .github/workflows/conformance.yml [hunk 2]
```diff
@@ -31,23 +31,36 @@ jobs:
...
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        config:
+          - name: "consul@v1.11.5 + consul-k8s@v0.44.0"
+            consul-image: "hashicorp/consul:1.11"
+            envoy-image: "envoyproxy/envoy:v1.20.2"
+            consul-k8s-version: "0.44.0"
+          - name: "consul@v1.12.2 + consul-k8s@v0.44.0"
+            consul-image: "hashicorp/consul:1.12"
```
[the remark this answers is not in the document]
Comment (sarahalsmiller) [reply .github/workflows/conformance.yml:42]: I'm assuming we'll have to keep this pin updated, but that doesn't seem like a big deal to me.

**Review by sarahalsmiller — APPROVED** [review]

File: .github/workflows/conformance.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nathancoleman) [reply .github/workflows/conformance.yml:42]: More likely that we'll add more configurations than update this one once consul 1.13 comes up until we break compatibility

**Pull request merged**